data/src/main/java/com/microsoft/azure/kusto/data/AadAuthenticationHelper.java

```diff
@@ -158,44 +165,107 @@ private AuthenticationResult acquireAccessTokenUsingDeviceCodeFlow() throws Exce
     }
 
     private AuthenticationResult waitAndAcquireTokenByDeviceCode(DeviceCode deviceCode, AuthenticationContext context)
-            throws  InterruptedException{
+            throws InterruptedException {
         int timeout = 15 * 1000;
         AuthenticationResult result = null;
-        while (timeout > 0){
-            try{
+        while (timeout > 0) {
+            try {
                 Future<AuthenticationResult> futureResult = context.acquireTokenByDeviceCode(deviceCode, null);
                 return futureResult.get();
             } catch (ExecutionException e) {
-                    Thread.sleep(1000);
-                    timeout -= 1000;
+                Thread.sleep(1000);
+                timeout -= 1000;
             }
         }
         return result;
     }
 
     AuthenticationResult acquireWithClientCertificate()
-            throws IOException, InterruptedException, ExecutionException, ServiceUnavailableException{
+            throws InterruptedException, ExecutionException, ServiceUnavailableException {
 
         AuthenticationContext context;
-        AuthenticationResult result;
+        AuthenticationResult result = null;
         ExecutorService service = null;
 
         try {
             service = Executors.newSingleThreadExecutor();
             context = new AuthenticationContext(aadAuthorityUri, false, service);
-            AsymmetricKeyCredential asymmetricKeyCredential = AsymmetricKeyCredential.create(clientCredential.getClientId(),
+            AsymmetricKeyCredential asymmetricKeyCredential = AsymmetricKeyCredential.create(applicationClientId,
                     privateKey, x509Certificate);
             // pass null value for optional callback function and acquire access token
             result = context.acquireToken(clusterUrl, asymmetricKeyCredential, null).get();
+        } catch (MalformedURLException e) {
+            e.printStackTrace();
         } finally {
             if (service != null) {
                 service.shutdown();
             }
         }
+
         if (result == null) {
             throw new ServiceUnavailableException("authentication result was null");
         }
         return result;
     }
 
+    private void acquireToken() throws DataServiceException {
```

Comment:
maybe rename to **acquireAccessToken** to better differentiate from the refresh token part?

Reply:
there is already an acquireAccessToken